### PR TITLE
Added refresh token support and fixed auth middleware

### PR DESCRIPTION
**Pull request by yash-chad**

### CODE REVIEW

File: infra/jwt.js [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
+	return jwt.sign(obj, jwtSecret, { expiresIn: 86400 });
+};
+
+exports.createRefreshToken = async obj => {
+	return jwt.sign(obj, jwtRefreshSecret, { expiresIn: 86400 * 7 });
 };
 
 exports.verifyJwt = async token => {
-	return jwt.verify(token, jwtSecret);
+	try{
```
Comment (siddinc) [inline infra/jwt.js]: - Avoid using try catch blocks for async await calls inside utility functions.
- Instead throw an error in the next() middleware and use a try catch block in the universal error handler

File: middleware/auth.js [hunk 2]
```diff
@@ -8,15 +8,14 @@ const { verifyJwt } = require('../infra/jwt');
  */
 const authRequired = async (req, res, next) => {
   const token = req.header('Authorization').replace('Bearer', '').trim();
-  const decoded = verifyJwt(token);
-  const user = await User.findOne({ _id: decoded._id, 'tokens.token': token });
-
-  if (!user) {
+  const decoded = await verifyJwt(token);
+  if(!decoded){
```
Comment (siddinc) [inline middleware/auth.js]: - Could instead throw and error in the next() middleware here if the token cannot be verified.
- Let the error handler send all the error responses to the client

File: infra/jwt.js (hunk 1 above)
Comment (yash-chad) [reply infra/jwt.js]: @siddinc I had to use a try/catch because verifyJwt should return some value which was then checked in the authRequired function

Comment (vixrant) [reply infra/jwt.js]: Since this function is highly reusable, let it throw an error. Handle errors where you're using it.

**Pull request merged**